plugins/net_plugin/net_plugin.cpp

```diff
@@ -2918,9 +2919,18 @@ namespace eosio {
             std::unique_lock<std::mutex> g( conn_mtx );
             const auto last_sent_lib = last_handshake_sent.last_irreversible_block_num;
             g.unlock();
+
             if( blk_num < last_sent_lib ) {
                fc_ilog( logger, "received block ${n} less than sent lib ${lib}", ("n", blk_num)("lib", last_sent_lib) );
-               close();
+               if(my_impl->p2p_always_resync) {
+                  fc_ilog( logger, "attempting to resync with peer that is sending blocks less than sent lib ${lib}", ("lib", last_sent_lib) );
+                  enqueue( (sync_request_message) {0, 0} );
+                  send_handshake();
+                  cancel_wait();
+               }
```

Comment:
I think we should not add this new option and just make this the behavior.

Reply:
Please put in your review comments and you can approve when it is done.